python/cugraph/community/triangle_count_wrapper.pyx

```diff
@@ -31,25 +30,24 @@ def triangles(input_graph):
     """
     Call triangle_count_nvgraph
     """
-    cdef uintptr_t graph = graph_wrapper.allocate_cpp_graph()
-    cdef Graph * g = <Graph*> graph
-
-    if input_graph.adjlist:
-        [offsets, indices] = graph_wrapper.datatype_cast([input_graph.adjlist.offsets, input_graph.adjlist.indices], [np.int32])
-        [weights] = graph_wrapper.datatype_cast([input_graph.adjlist.weights], [np.float32, np.float64])
-        graph_wrapper.add_adj_list(graph, offsets, indices, weights)
-    else:
-        [src, dst] = graph_wrapper.datatype_cast([input_graph.edgelist.edgelist_df['src'], input_graph.edgelist.edgelist_df['dst']], [np.int32])
-        if input_graph.edgelist.weights:
-            [weights] = graph_wrapper.datatype_cast([input_graph.edgelist.edgelist_df['weights']], [np.float32, np.float64])
-            graph_wrapper.add_edge_list(graph, src, dst, weights)
-        else:
-            graph_wrapper.add_edge_list(graph, src, dst)
-        add_adj_list(g)
-        offsets, indices, values = graph_wrapper.get_adj_list(graph)
-        input_graph.adjlist = input_graph.AdjList(offsets, indices, values)
-
-    cdef uint64_t result
-    triangle_count_nvgraph(g, &result)
+    offsets = None
+    indices = None
+
+    if not input_graph.adjlist:
+        input_graph.view_adj_list()
+
+    [offsets, indices] = graph_new_wrapper.datatype_cast([input_graph.adjlist.offsets,
+                                                          input_graph.adjlist.indices], [np.int32])
+
+    num_verts = input_graph.number_of_vertices()
+    num_edges = len(indices)
+
+    cdef uintptr_t c_offsets = offsets.__cuda_array_interface__['data'][0]
+    cdef uintptr_t c_indices = indices.__cuda_array_interface__['data'][0]
+
+    cdef GraphCSR[int,int,float] graph
```

Comment:
Random general question: 
we don't need weights (or weights types) here. Is there a way we could just declare a `GraphCSR[int,int]` so that we avoid dealing with the dummy `float` type in unweighted analytics wrappers? We could leverage default template parameters at the C++ level so that the graph class implementation could just default to a null float*, but not sure if cython would support that. Thoughts?

Reply:
I thought about that when I was constructing the objects.  I don't know cython well enough to know if it would support defaulted parameters.  It seemed an innocuous amount of extra typing to specify float.

The underlying C++ classes allow the pointer to be null and just ignore it.  There's no real penalty other than having to specify a type of something that's unused.